### PR TITLE
more edits

### PR DESCRIPTION
**Pull request by petrelharp**
I'll start compillng my new edits here (and will rebase after Andy's gets merged).

### CODE REVIEW

File: ms/spaceness.tex [hunk 1]
```diff
@@ -256,7 +256,9 @@ \subsection{Demographic Parameters}
 in a way that depends on the neighborhood size.
 Therefore, to properly compare to theoretical expectations,
 we empirically calculated these demographic quantities in simulations. 
-We recorded the census population size in all simulations. 
+We recorded the census population size in all simulations,
+and used mean population density ($\rho$, census size divided by total area)
+to compute neighborhood size as $N_W = 4 \pi \rho \sigma^2$. \revpoint{3}{6}
```
Comment (petrelharp) [inline ms/spaceness.tex]: saying it explicitly here

File: ms/spaceness.tex [hunk 2]
```diff
@@ -374,7 +376,12 @@ \subsection{Demographic Parameters and Run Times}
...
+Census sizes range from around 14,000 at low $\sigma$ in the random mating model to 10,000 for both models when neighborhood sizes approach 1,000.
+The scaling of census sizes in both random-mating and spatial models appears to be related to two consequences of the spatial competition function: the decline of fitness at range edges, which effectively reduces the habitable area by one $\sigma$ around the edge of the map and so results in a smaller habitable area at high $\sigma$ values; and variation in the equilibrium population density given varying competition radii.
+Furthermore, lack of available mates contributes to lower census sizes at small $N_W$
+for the spatial model
+(We calculate $N_W$ for both in the same way,
+even though the number of ``available mates'' is much larger with random mating). \llabel{randomNW}
```
Comment (petrelharp) [inline ms/spaceness.tex]: making the point about lack of mates also

File: ms/spaceness.tex [hunk 3]
```diff
@@ -511,7 +518,9 @@ \subsection{GWAS}
...
+    \caption{Impacts of spatially varying environments and isolation by distance on linear regression GWAS. Simulated quantitative phenotypes are determined only by an individual's location and the spatial distribution of environmental factors. In \textbf{A} we show the phenotypes and locations of sampled individuals under four environmental distributions, with transparency scaled to phenotype. As neighborhood size increases a PCA explains less of the total variation in the data (\textbf{B}).
+    Spatially correlated environmental factors cause false positives at a large proportion of SNPs, which is partially but not entirely corrected by adding the first 10 PC coordinates as covariates (\textbf{C}).
+    Quantile-quantile plots in (\textbf{D}) show inflation of $-\log_{10}(p)$ after PC correction for simulations with spatially structured environments, with line colors showing the neighborhood size of each simulation.
```
Comment (petrelharp) [inline ms/spaceness.tex]: my only actual change here was "line colors scaled to" -> "line colors showing".
Diff doesn't highlight changes on these very long lines.

File: ms/Makefile [hunk 4]
```diff
@@ -39,5 +39,5 @@ clean:
 
 spaceness_diff_to_first_submission.tex:
```
Comment (petrelharp) [inline ms/Makefile]: you can do `make spaceness_diff_to_first_submission.pdf` to get the latexdiff to the previous submission (and should include this pdf in the resubmission).

File: ms/review-responses.tex [hunk 5]
```diff
@@ -25,7 +25,17 @@
...
+which we have addressed in detail, below.
+The main changes we have made have to do with our choice of model.
+We have added some more discussion
+of the biological motivation for the particular model of continuous space we use
+(with references to the ecological literature, where it appears) to the Methods (around \llname{ecolcites}),
+and in the Discussion (at \llname{demog_disco}).
+We have included more explanation of some of the more opaque consequences
+of a more realistic demographic model, around \llname{censusscaling}.
+Finally, we have added a substantial comparison to the discrete stepping-stone model,
+most of which is in the Appendix.
```
Comment (petrelharp) [inline ms/review-responses.tex]: I think these are the major things; anything else we want to mention? Should also say here that we're including a pdf highlighting differences (the latexdiff).

File: ms/review-responses.tex [hunk 6]
```diff
@@ -592,14 +601,18 @@
 
 \reply{
    We have expanded our citations of some of the relevant ecology literature \llname{ecolcites}, which we hope helps to contextualize the study better.
+   We haven't cited dadi, which works with discrete populations
+   (the ``diffusion'' is in frequency space).
```
Comment (petrelharp) [inline ms/review-responses.tex]: right? they are totally misunderstanding dadi?

File: ms/review-responses.tex [hunk 7]
```diff
@@ -592,14 +601,18 @@
...
 
 \begin{point}{}
     There is some reference to spatial models using grids (e.g. Rousset 1997). Some additional discussion contextualizing more recent methods like EEMS that also construct demes and model migration through divergence between neighboring demes would be helpful and interesting.
 \end{point}
 
 \reply{
-    We agree that not noting EEMS was an oversight. We have added the most recent EEMS paper to the citations on \llname{eemscitation}, and have added an Appendix giving a more thorough comparison with stepping-stone models. 
+    We agree that not noting EEMS was an oversight.
+    We have added the most recent EEMS paper to the citations at \revref,
+    and have added an Appendix giving a more thorough comparison with stepping-stone models.
```
Comment (petrelharp) [inline ms/review-responses.tex]: you had actually cited MAPS; I added the ref to EEMS (and, actually, these both integrate over a bunch of discrete models, so they are a little different...)

File: ms/review-responses.tex [hunk 8]
```diff
@@ -619,27 +632,28 @@
 }
 
 \begin{point}{}
-    Code availability. This github link doesn't work, but is important to be able to evaluate for review: https://github.com/petrelharp/spaceness
+    Code availability. This github link doesn't work, but is important to be able to evaluate for review: \url{https://github.com/petrelharp/spaceness}.
```
Comment (petrelharp) [inline ms/review-responses.tex]: whoops! This should be `cjbattey`, right? Unless you want mine to be the official repo? (And, to make your repo not a fork of mine you need to delete/rename it and start a new one, btw.)

File: ms/review-responses.tex [hunk 9]
```diff
@@ -619,27 +632,28 @@
...
 }
 
 \begin{point}{}
-    Definitions and interpretations. There are quite a large number of metrics discussed in Figure 3B, and it's a lot to take in. It might be helpful to have a table with a reminder of what the metric is, its interpretation, and how it is computed.
+    Definitions and interpretations. There are quite a large number of metrics discussed in Figure \ref{fig:sumstats}, and it's a lot to take in. It might be helpful to have a table with a reminder of what the metric is, its interpretation, and how it is computed.
 \end{point}
 
 \reply{
-    We have included a table describing the summary statistics in Figure \ref{table:sumstats}.
+    We have included a table describing the summary statistics in \autoref{table:sumstats}.
```
Comment (petrelharp) [inline ms/review-responses.tex]: table, not figure

File: ms/review-responses.tex [hunk 10]
```diff
@@ -619,27 +632,28 @@
...
 }
 
-\begin{point}{}
+\begin{point}{\revref}
     Notation: ``Offspring disperse a Gaussian-distributed distance away from the parent with mean zero and standard deviation $\sigma$ in both the x and y coordinates. Each offspring is produced with a mate selected randomly from those within distance $3\sigma$, with probability of choosing a neighbor at distance $x$ proportional to $\exp(-x^2/2\sigma^2)$.'' I think x may be overloaded here, or I'm confused. Clarify?
 \end{point}
 
 \reply{
-	We have clarified the notation for the second instance (referring to the distance among individuals) to $d$. 
+	We have clarified the notation for the second instance (referring to the distance among individuals) to $d$. \revref
```
Comment (petrelharp) [inline ms/review-responses.tex]: just adding line references

File: ms/spaceness.tex [hunk 11]
```diff
@@ -244,7 +245,16 @@ \subsection{A Forward-Time Model of Evolution in Continuous Space}
 as predicted by theory \citep{ralph2019efficiently}.
 Simulations were run for 1.6 million timesteps (approximately $30N$ generations). 
 
-To check that our model produces reasonable results, we compared its output to that of a stepping-stone model implemented in msprime \citep{Kelleher2016}. These results are discussed in detail in Appendix 1, but in general we find that our model produces many of patterns generated by stepping stone models while avoiding some artifacts associated with discretization of the landscape.
```
Comment (petrelharp) [inline ms/spaceness.tex]: We're not checking *our* model, we're checking whether the grid model is reasonable! And, I don't think that the general message is that "they agree", which is what yours sounded like; do you agree with what I've written?

File: ms/spaceness.tex [hunk 12]
```diff
@@ -285,16 +295,17 @@ \subsection{Summary Statistics}
...
 which we summarized as the correlation coefficient between the logarithm of the spatial distance and the proportion of identical base pairs across pairs of individuals. 
 
-Following recent studies that showed strong signals for dispersal and demography in the distribution of shared haplotype block lengths \citep{Ringbauer2017,Baharian2016}, we also calculated various summaries of the distribution of pairwise identical-by-state (IBS) block lengths among sampled chromosomes. The full distribution of lengths of IBS tracts for each pair of chromosomes was first calculated with a custom python function. 
-%\url{https://github.com/petrelharp/spaceness/blob/master/scripts/slimtools.py}. 
+Following recent studies that showed strong signals for dispersal and demography in the distribution of shared haplotype block lengths \citep{Ringbauer2017,Baharian2016}, we also calculated various summaries of the distribution of pairwise identical-by-state (IBS) block lengths among sampled chromosomes,
+defined to be the set of distances between adjacent sites that differ between the two chromosomes
```
Comment (petrelharp) [inline ms/spaceness.tex]: just defining it to be totally sure

File: ms/spaceness.tex [hunk 13]
```diff
@@ -285,16 +295,17 @@ \subsection{Summary Statistics}
...
 
 The effects of sampling on summary statistic estimates were summarized by testing for differences in mean (ANOVA, \citep{Rcore2018}) and variance (Levene's test, \citep{Fox2011}) across sampling strategies for each summary statistic. 
 
-Last, we used a subset of summary statistics to compare how our continuous model performs relative to a reverse-time stepping stone model (Appendix 1). 
-
```
Comment (petrelharp) [inline ms/spaceness.tex]: already said this; I think we can skip this

File: ms/spaceness.tex [hunk 14]
```diff
@@ -605,7 +616,9 @@ \subsection{Demography}
...
+We found that SMC++ was more robust to the effects of space than stairwayplot, underestimating recent populations by roughly half in the worst time periods rather than nearly 10-fold as with stairwayplot. Though this degree of variation in population size is certainly meaningful in an ecological context, it is relatively minor in population genetic terms. \llabel{ibdnesxn}
+Methods directly assessing haplotype structure in phased data (for example, \citet{browning2015accurate}) are thought to provide increased resolution for recent demographic events,
+but in this case the error we observed was essentially an accurate reflection of underlying genealogies in which terminal branches are anomalously short. Combined with our analysis of IBS tract length variation (Figure \ref{fig:ibs}) this suggests that haplotype-based methods are likely to be affected by similar biases.
```
Comment (petrelharp) [inline ms/spaceness.tex]: this was just minor word changes

File: ms/spaceness.tex [hunk 15]
```diff
@@ -790,7 +803,20 @@ \section{Acknowledgements}
...
...}. In this class of models we imagine an $n \times n$ grid of populations exchanging migrants with neighboring populations at rate $m$. If these models are good approximations of the continuous case we expect that results will converge as $n \to \infty$, so we ran simulations while varying $n$ from 5 to 50 (Table \ref{table:grid_params}). To compare with continuous models we first distributed the the same "effective" number of individuals across the landscape as in our continuous-space simulations ($\approx 6100$, estimated from $\theta_{\pi}$ of random-mating continuous-space simulations). We then approximate the mean per-generation dispersal distance $\sigma$ given a total landscape width $W$ as the product of the probability of an individual being a migrant and the distance traveled by migrants: $\sigma \approx 4m(W/n)$. We ran 500 simulations per $n$ while sampling $\sigma$ from $U(0.2,4)$. We then randomly selected 60 diploid individuals from each simulation (approximating diploidy by combining pairs of chromosomes with contiguous indices within demes) and calculated a set of six summary statistics using the scripts described in the summary statistics portion of the main text.
```
Comment (petrelharp) [inline ms/spaceness.tex]: as above. We should probably have a bit more explanation of the patterns, here?

**Comment by petrelharp** [comment]
I've gone through all the responses, and want to do a quick pass through the thing from the top; I'll keep pushing to this PR so we can discuss here?

File: ms/spaceness.tex (hunk 2 above)
Comment (andrewkern) [reply ms/spaceness.tex]: why is "available mates" in double quotes the second time but not the first?

File: ms/review-responses.tex (hunk 6 above)
Comment (andrewkern) [reply ms/review-responses.tex]: yeah i have no idea what's going on there. we should also add that dadi isn't suited for more than 3 populations, so even stepping stone models are quite limited

File: ms/review-responses.tex [hunk 16]
```diff
@@ -609,27 +632,28 @@
...
 
 \begin{point}{}
-    Code availability. This github link doesn't work, but is important to be able to evaluate for review: https://github.com/petrelharp/spaceness
+    Code availability. This github link doesn't work, but is important to be able to evaluate for review: \url{https://github.com/petrelharp/spaceness}.
 \end{point}
 
 \reply{
-Apologies, it was accidentally set to private. The link should work now. 
+Apologies, it was accidentally set to private. The link should work now
+(and has been replaced by the primary link, \url{https://github.com/cjbattey/spaceness}).
```
Comment (andrewkern) [inline ms/review-responses.tex]: @cjbattey  let's fork the current version to https://github.com/kern-lab/ and that can take over as the official release. we could even use the Github releases feature at this point.

Comment (cjbattey) [reply ms/review-responses.tex]: just merged this PR so the repo should be ready to fork. Should we change the links in the ms  after forking or before?

File: ms/spaceness.tex (hunk 2 above)
Comment (petrelharp) [reply ms/spaceness.tex]: oh good point

File: ms/review-responses.tex (hunk 6 above)
Comment (petrelharp) [reply ms/review-responses.tex]: meh, no need to rub it in

File: ms/spaceness.tex (hunk 15 above)
Comment (cjbattey) [reply ms/spaceness.tex]: sorry missed this before merging. What did you have in mind for more explanation of patterns? There is some in the next paragraph

File: ms/spaceness.tex (hunk 11 above)
Comment (petrelharp) [reply ms/spaceness.tex]: see stuff I wrote here, @cjbattey

File: ms/review-responses.tex (hunk 16 above)
Comment (andrewkern) [reply ms/review-responses.tex]: let's change them after forking to make sure the address is set. i'll fork it now.

File: ms/spaceness.tex (hunk 11 above)
Comment (cjbattey) [reply ms/spaceness.tex]: oh I see now. I think this is a fair summary and don't mind the tonal shift (I had in mind just that pi and geographic distance generally decline as dispersal increases, which I take to be the general patterns people think of as isolation by distance)

Comment (petrelharp) [reply ms/spaceness.tex]: Ah, good point that it's at least going in the same direction. I think we should put some more nuanced discussion along these lines in the Appendix? How about we talk through the comparison in person tomorrow and then write a few more paragraphs on the basis of that?

Comment (cjbattey) [reply ms/spaceness.tex]: sure, sounds good

**Pull request merged**